manim/utils/iterables.py

```diff
@@ -99,13 +93,7 @@ def tuplify(obj):
 
 
 def batch_by_property(items, property_func):
-    """
-    Takes in a list, and returns a list of tuples, (batch, prop)
-    such that all items in a batch have the same output when
-    put into property_func, and such that chaining all these
-    batches together would give the original list (i.e. order is
-    preserved)
-    """
+    """Takes in a list, and returns a list of tuples, (batch, prop) such that all items in a batch have the same output when put into property_func, and such that chaining all these batches together would give the original list (i.e. order is preserved)"""
```

Comment:
Long line. Not sure how to shorten, but this seems to be conceptually the same as [itertools.groupby](https://docs.python.org/3/library/itertools.html#itertools.groupby).

Reply:
Maybe open an issue about that, could be an optimization.

Reply:
```suggestion
    """Group all items in the list with their properties into tuples (batch, property)
    
    Takes in a list, and returns a list of tuples, (batch, prop)
    such that all items in a batch have the same output when
    put into property_func, and such that chaining all these
    batches together would give the original list (i.e. order is
    preserved)
    """
```
Will this do?